### PR TITLE
Fixes issue where relativity would not connect via Octopus gem.

### PR DESCRIPTION
**Pull request by michaelachrisco**

### CODE REVIEW

**Comment by michaelachrisco** [comment]
@josephbridgwaterrowe Please review.

File: app/models/driver_adapter.rb [hunk 1]
```diff
@@ -9,20 +10,33 @@ class DriverAdapter
 
   class << self
     def connection
-      ActiveRecord::Base.using(:grossman).connection
+      Octopus.using(:grossman) do
```
Comment (josephbridgwaterrowe) [inline app/models/driver_adapter.rb]: We discussed refactoring these common methods into a base class. Something like: 

```ruby
GrossmanAdapter < ActiveRecord::Base
  class << self
    def connection
      Octopus.using(:grossman) do
        ActiveRecord::Base.connection
      end
    end

    def connected_with_octopus?
      Octopus.using(:grossman) do
        ActiveRecord::Base.connected_with_octopus?
      end
    end

    def not_supported
      fail 'Not supported!'
    end
    alias :new :not_supported
    alias :delete :not_supported
    alias :destroy :not_supported
  end
end
```

Also, can't you remove the `Octopus.using(:grossman) do` by using a `super.using(:grossman)` at the top of the class?

At which point you could completely drop the connection method from self.

So, in theory:

```ruby
# grossman_adapter.rb
GrossmanAdapter < ActiveRecord::Base
  self.using(:grossman)

  class << self
    def not_supported
      fail 'Not supported!'
    end
    alias :new :not_supported
  end

  def not_supported
    GrossmanAdapter.not_supported
  end
  alias :delete, :not_supported
  alias :destroy, :not_supported

  def sanitize_sql(sql)
    ActiveRecord::Base.send(:sanitize_sql_array, sql)
  end
end

# driver_adapter.rb
class DriverAdapter < GrossmanAdapter
  class << self
    def all
      results = connection.execute(sanitize_sql([QUERY, 'DR']))
      results.map(&:to_snake_keys)
    end
  end
end
```

I think you might want to look at the alias as some of those methods would be instance, some class.

_The alias example are guess work, not sure if they would work._

File: .gitignore [hunk 2]
```diff
@@ -36,6 +36,7 @@ spec/cassettes
 
 # Ignore application configuration
 /config/application.yml
+config/shards.yml.ig2
```
Comment (josephbridgwaterrowe) [inline .gitignore]: :gun:

File: app/models/driver_commission_history_adapter.rb [hunk 3]
```diff
@@ -1,6 +1,7 @@
 # DriverCommissionHistoryAdapter
-class DriverCommissionHistoryAdapter
+class DriverCommissionHistoryAdapter < GrossmanAdapter
   # self.using(:grossman).connection.current_shard
```
Comment (josephbridgwaterrowe) [inline app/models/driver_commission_history_adapter.rb]: :gun:

File: app/models/grossman_adapter.rb [hunk 4]
```diff
@@ -0,0 +1,24 @@
+class GrossmanAdapter < ActiveRecord::Base
+  # self.using(:grossman).connection.current_shard
```
Comment (josephbridgwaterrowe) [inline app/models/grossman_adapter.rb]: :gun:

File: app/models/inventory_item_adapter.rb [hunk 5]
```diff
@@ -1,6 +1,7 @@
 # DriverCommissionHistoryAdapter
-class InventoryItemAdapter
+class InventoryItemAdapter  < GrossmanAdapter
   # self.using(:grossman).connection.current_shard
+
```
Comment (josephbridgwaterrowe) [inline app/models/inventory_item_adapter.rb]: :gun:

File: spec/models/grossman_adapter_spec.rb [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    it { expect { subject }.to raise_error.with_message(/Not supported!/) }
+  end
+
+  describe '#destroy' do
+    subject { gman_adapter.destroy }
+
+    it { expect { subject }.to raise_error.with_message(/Not supported!/) }
+  end
+
+  describe '#gman_connection' do
```
Comment (josephbridgwaterrowe) [inline spec/models/grossman_adapter_spec.rb]: current_shard?

File: spec/models/grossman_adapter_spec.rb [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+    subject do
+      gman_adapter.gman_connection.current_shard
+    end
+
+    it 'current_shard is equal to grossman' do
+      expect(is_expected.to eq(:grossman))
```
Comment (josephbridgwaterrowe) [inline spec/models/grossman_adapter_spec.rb]: `expect(current_shard).to eq(:grossman)`

And WTF? :)

File: spec/models/grossman_adapter_spec.rb (hunk 6 above)
Comment (michaelachrisco) [inline spec/models/grossman_adapter_spec.rb]: What do you mean? If
```Ruby
  describe '#gman_connection' do
    subject do
      gman_adapter.gman_connection
    end

    it 'current_shard is equal to grossman' do
      expect(is_expected.to eq(:grossman))
    end
  end
```
Then the test will fail.

Comment (josephbridgwaterrowe) [inline spec/models/grossman_adapter_spec.rb]: See below for the expectation... :)

File: spec/models/grossman_adapter_spec.rb (hunk 7 above)
Comment (michaelachrisco) [inline spec/models/grossman_adapter_spec.rb]: I am still confused here.

Comment (josephbridgwaterrowe) [inline spec/models/grossman_adapter_spec.rb]:
```ruby
  describe '#gman_connection' do
    subject do
      gman_adapter.gman_connection
    end

    it 'current_shard is equal to grossman' do
      expect(subject.current_shard).to eq(:grossman)
    end
  end
```

**Comment by michaelachrisco** [comment]
@josephbridgwaterrowe Review and merge?

**Pull request merged**